### PR TITLE
Update flyway to use Serde and latest Milestones and RCs for 4.0.0

### PR DESCRIPTION
**Pull request by timyates**
Updated Flyway to latest version.
Switched to using Serde instead of Jackson.
Updated to latest RCs and Milestones of micronaut modules. Use Typesafe project accessors.

### CODE REVIEW

File: flyway/build.gradle [hunk 1]
```diff
@@ -4,8 +4,10 @@ plugins {
 
 dependencies {
     annotationProcessor mn.micronaut.graal
+    annotationProcessor mnSerde.micronaut.serde.processor
 
     compileOnly libs.graal.svm
+    compileOnly mnSerde.micronaut.serde.api
```
Comment (sdelamo) [inline flyway/build.gradle:10]: should not `mnSerde.micronaut.serde.api` be `api` instead of `compileOnly`?

**Review by sdelamo — APPROVED** [review]

File: flyway/build.gradle (hunk 1 above)
Comment (timyates) [reply flyway/build.gradle:10]: Wouldn't that mean that we pull the serde API library into someone's classpath if they use flyway? Even if they want jackson-databind?

Comment (sdelamo) [reply flyway/build.gradle:10]: yes, but that will happen always if you want to expose an api and your API classes need serde api annotations.

Comment (timyates) [reply flyway/build.gradle:10]: But people using this don't need Serde do they?  Isn't it a placeholder to generate classes in this module at compile-time?

[I was following what we do with micronaut-data](https://github.com/micronaut-projects/micronaut-data/blob/f15a90715b5e0c29ffcde67e47aa98b5f83b2b17/data-model/build.gradle)

Comment (timyates) [reply flyway/build.gradle:10]: Another datapoint 😉 https://github.com/micronaut-projects/micronaut-liquibase/blob/master/liquibase/build.gradle

**Pull request merged**